js/src/@types/global.d.ts

```diff
@@ -1,3 +1,31 @@
+/**
+ * A type that matches any valid value for the `type` attribute on an
+ * HTML `<input>` element.
+ */
+declare type HTMLInputTypes =
```

Comment:
Why?

Reply:
I considered it to be more of a general type as opposed to being specific to that component. There's a good chance we might want to use it elsewhere (or another Flarum ext might). It's mainly for convenience.

Reply:
Maybe we should move it into the [generic form builder](https://github.com/flarum/core/pull/2996#discussion_r691807155) when we have that? I'm hesitant to put stuff in global that doesn't absolutely need to be there.